app/src/main/java/pl/droidevs/books/login/LoginActivity.java

```diff
@@ -1,37 +1,87 @@
 package pl.droidevs.books.login;
 
+import android.arch.lifecycle.ViewModelProvider;
+import android.arch.lifecycle.ViewModelProviders;
 import android.content.Intent;
 import android.os.Bundle;
 import android.support.v7.app.AppCompatActivity;
-import android.widget.TextView;
+import android.widget.Button;
+import android.widget.EditText;
+
+import javax.inject.Inject;
 
 import butterknife.BindView;
 import butterknife.ButterKnife;
 import butterknife.OnClick;
+import butterknife.OnTextChanged;
 import butterknife.Unbinder;
 import dagger.android.AndroidInjection;
 import pl.droidevs.books.R;
 import pl.droidevs.books.library.LibraryActivity;
 
 public class LoginActivity extends AppCompatActivity {
 
-    @BindView(R.id.textHello)
-    TextView textHello;
+    @BindView(R.id.login_edit_text)
+    EditText loginEditText;
+
+    @BindView(R.id.login_button)
+    Button loginButton;
 
     private Unbinder unbinder;
 
+    @Inject
+    ViewModelProvider.Factory viewModelFactory;
+
+    private LoginViewModel loginViewModel;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_login);
 
         AndroidInjection.inject(this);
         this.unbinder = ButterKnife.bind(this);
```

Comment:
Wiem, że tego teraz nie zmieniałaś, ale może warto to przy okazji też posprzątać. Otóż z tego co wyczytałem unbind jest potrzebne tylko przy fragmentach: https://stackoverflow.com/questions/38015412/where-should-i-unbind-butterknife-8-x-x-in-a-viewholder

Reply:
Dzięki za źródło